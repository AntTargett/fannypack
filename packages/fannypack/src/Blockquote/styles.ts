import { darken } from 'polished';
import { cssClass, space, theme } from '../styled';

export const Blockquote = styleProps => cssClass`
  border-left: 4px solid ${darken(0.2, 'white')};
  padding: ${space(4)(styleProps)}rem;

  &:not(:last-child) {
    margin-bottom: ${space(4)(styleProps)}rem;
  }

  & {
    ${theme('Blockquote.base')(styleProps)};
  }
`;
