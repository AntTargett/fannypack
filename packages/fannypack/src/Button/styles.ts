import { darken } from 'polished';
import getDefaultPalette from '../theme/palette';
import { css, cssClass, palette, space, theme } from '../styled';

const defaultPalette = getDefaultPalette({});

export const Button = styleProps => cssClass`
  align-items: center;
  background-color: ${palette(styleProps.palette, 'red')(styleProps)};
  border-radius: 4px;
  color: ${palette(`${styleProps.palette}Inverted`, 'red')(styleProps)};
  fill: ${palette(`${styleProps.palette}Inverted`, 'red')(styleProps)};
  cursor: pointer;
  display: inline-flex;
  font-weight: ${theme('fontWeights.semibold')(styleProps)};
  min-height: 2.5em;
  justify-content: center;
  padding: 0 0.8rem;
  text-decoration: none;
  hyphens: auto;
  transition: box-shadow 0.1s ease-in-out 0s, border 0.1s, background-color 0.1s;

  ${styleProps.palette === 'default' &&
    css`
      border: 1px solid ${palette('gray100')(styleProps)};
    `};

  & {
    ${theme('Button.base')(styleProps)};
  }

  &[disabled] {
    ${getDisabledProperties(styleProps)};
  }

  &:focus {
    outline: unset;
    z-index: 2;
    box-shadow: ${palette(styleProps.palette === 'default' ? 'primary' : `${styleProps.palette}300`)(
      styleProps
    )} 0px 0px 0px
      2px;

    ${styleProps.palette === 'default' &&
      css`
        border-color: transparent;
      `}

    ${theme('Button.focus')(styleProps)};
  }

  ${styleProps.size && getSizeProperties(styleProps)}
  ${styleProps.isLoading && getLoadingProperties(styleProps)};
  ${styleProps.isStatic && getStaticProperties(styleProps)};
  ${isInteractive(styleProps) && getInteractiveProperties(styleProps)};

  ${styleProps.kind === 'outlined' && getOutlinedProperties(styleProps)};
  ${styleProps.kind === 'link' && getLinkProperties(styleProps)};
  ${styleProps.kind === 'ghost' && getGhostProperties(styleProps)};
`;

export const isInteractive = styleProps =>
  !styleProps.isStatic && !styleProps.isLoading && !styleProps.disabled && styleProps.kind !== 'link';

export const getDisabledProperties = styleProps => css`
  & {
    cursor: not-allowed;
    opacity: 0.7;
    outline: unset;
    pointer-events: unset;
  }
  & {
    ${theme('Button.disabled')(styleProps)};
  }
`;

export const getSizeProperties = styleProps => {
  const styles = {
    small: css`
      & {
        font-size: ${theme('fontSizes.100')(styleProps)}em;
        min-height: ${space(8)(styleProps)}em;
        padding: 0 ${space(2)(styleProps)}rem;
      }
      & {
        ${theme('Button.sizes.small')(styleProps)};
      }
    `,
    default: css`
      & {
        ${theme('Button.sizes.default')(styleProps)};
      }
    `,
    medium: css`
      & {
        min-height: ${space(12)(styleProps)}em;
        padding: 0 ${space(5)(styleProps)}rem;
      }
      & {
        ${theme('Button.sizes.medium')(styleProps)};
      }
    `,
    large: css`
      & {
        font-size: ${theme('fontSizes.300')(styleProps)}em;
        min-height: ${space(12)(styleProps)}em;
        padding: 0 ${space(6)(styleProps)}rem;
      }
      & {
        ${theme('Button.sizes.large')(styleProps)};
      }
    `
  };
  return styles[styleProps.size];
};

export const getLoadingProperties = styleProps => css`
  & {
    cursor: not-allowed;
    position: relative;

    &:focus {
      box-shadow: unset !important;
      outline: unset !important;
    }
  }
  & {
    ${theme('Button.loading')(styleProps)};
  }
`;

export const getStaticProperties = styleProps => css`
  & {
    cursor: default;
    outline: unset;

    &:focus {
      box-shadow: unset !important;
      outline: unset !important;
    }
  }
  & {
    ${theme('Button.static')(styleProps)};
  }
`;

export const getInteractiveProperties = styleProps => css`
  &:hover {
    background-color: ${darken(0.05, palette(styleProps.palette, 'red')(styleProps))};
    & {
      ${theme('Button.hover')(styleProps)};
    }
  }
  &:hover:active {
    background-color: ${darken(0.1, palette(styleProps.palette, 'red')(styleProps))};
    & {
      ${theme('Button.hoveractive')(styleProps)};
    }
  }
`;

export const getLinkProperties = styleProps => css`
  & {
    border: 0;
    background: unset;
    color: ${styleProps.palette === 'default'
      ? palette('text', defaultPalette.text)(styleProps)
      : palette(styleProps.palette)(styleProps)};
    fill: ${styleProps.palette === 'default'
      ? palette('text', defaultPalette.text)(styleProps)
      : palette(styleProps.palette)(styleProps)};
    text-decoration: underline;

    &:hover {
      color: ${styleProps.palette === 'default'
        ? darken(0.5, palette('text', defaultPalette.text)(styleProps))
        : darken(0.5, palette(styleProps.palette)(styleProps))};
      fill: ${styleProps.palette === 'default'
        ? darken(0.5, palette('text', defaultPalette.text)(styleProps))
        : darken(0.5, palette(styleProps.palette)(styleProps))};
    }
  }

  & {
    ${theme('Button.link')(styleProps)};
  }
`;

export const getOutlinedProperties = styleProps => css`
  & {
    background-color: unset;
    border: 1px solid ${palette()(styleProps)};
    color: ${palette()(styleProps)};
    fill: ${palette()(styleProps)};

    ${isInteractive(styleProps) &&
      css`
        &:hover {
          background-color: ${palette()(styleProps)};
          color: ${palette(`${styleProps.palette}Inverted`)(styleProps)};
          fill: ${palette(`${styleProps.palette}Inverted`)(styleProps)};
        }
      `};
  }
  & {
    ${theme('Button.outlined')(styleProps)};
  }
`;

export const getGhostProperties = styleProps => css`
  & {
    background-color: unset;
    border: unset;
    color: ${styleProps.palette === 'default' ? palette('defaultInverted')(styleProps) : palette()(styleProps)};
    fill: ${styleProps.palette === 'default' ? palette('defaultInverted')(styleProps) : palette()(styleProps)};

    &:hover {
      background-color: ${darken(0.05, palette('default')(styleProps))};
    }
    &:hover:active {
      background-color: ${darken(0.1, palette('default')(styleProps))};
    }
  }
  & {
    ${theme('Button.ghost')(styleProps)};
  }
`;
