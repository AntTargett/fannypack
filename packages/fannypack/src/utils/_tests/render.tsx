import * as React from 'react';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (Component: any, { theme = {} } = {}) => render(<Provider theme={theme}>{Component}</Provider>);
