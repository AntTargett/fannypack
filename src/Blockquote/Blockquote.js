// @flow
import React, { type Node } from 'react';
import _Blockquote from 'reakit/Blockquote';

import type { Palette } from '../types';

type Props = {
  children: Node,
  className?: string,
  palette?: Palette
};

const Blockquote = ({ children, className, ...props }: Props) => (
  <_Blockquote className={className} {...props}>
    {children}
  </_Blockquote>
);

Blockquote.defaultProps = {
  className: null,
  palette: null
};

export default Blockquote;
