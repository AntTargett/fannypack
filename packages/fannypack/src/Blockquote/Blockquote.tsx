import { Box as ReakitBox } from 'reakit';

import { BlockquoteThemeConfig } from '../types';
import { useClassName, createComponent, createElement } from '../utils';
import { Box, BoxProps } from '../Box';

import * as styles from './styles';

export type LocalBlockquoteProps = {
  overrides?: BlockquoteThemeConfig;
};
export type BlockquoteProps = BoxProps & LocalBlockquoteProps;

function useProps(props: Partial<BlockquoteProps> = {}) {
  const boxProps = Box.useProps(props);

  const className = useClassName({
    style: styles.Blockquote,
    styleProps: props,
    prevClassName: boxProps.className
  });

  return { ...boxProps, className };
}

export const Blockquote = createComponent<BlockquoteProps>(
  props => {
    const { children, use, ...restProps } = props;
    const blockquoteProps = useProps(restProps);
    return createElement({ children, component: ReakitBox, use, htmlProps: blockquoteProps });
  },
  { defaultProps: { use: 'blockquote' }, useProps }
);
