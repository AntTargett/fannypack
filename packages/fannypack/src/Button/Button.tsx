import * as React from 'react';
import { Button as ReakitButton, useButton as useReakitButton } from 'reakit';

import * as utils from '../utils';
import { ButtonThemeConfig, ButtonKind, ButtonType, Size, Palette } from '../types';
import { BoxProps, useBoxProps } from '../Box';

import * as styles from './styles';

export type LocalButtonProps = {
  /** Makes the button disabled. The user is unable to interact with the button. */
  disabled?: boolean;
  /** TODO: Icon that appears on the right side of the button. */
  iconAfter?: any;
  /** TODO: Icon that appears on the left side of the button. */
  iconBefore?: any;
  /** TODO: Adds a loading indicator to the button. */
  isLoading?: boolean;
  /** Makes the button not interactable. */
  isStatic?: boolean;
  kind?: ButtonKind;
  /** Theming overrides. [Click here](TODO) to view the theming schema for `Button`. */
  overrides?: ButtonThemeConfig;
  palette?: Palette;
  size?: Size;
  type?: ButtonType;
};
export type ButtonProps = BoxProps & LocalButtonProps;

Button.defaultProps = {
  disabled: false,
  iconAfter: undefined,
  iconBefore: undefined,
  isLoading: false,
  isStatic: false,
  kind: undefined,
  palette: 'default',
  size: 'default',
  type: 'button'
};

export function useButtonProps(props: Partial<ButtonProps> = {}) {
  const boxProps = useBoxProps(props);
  const buttonProps = useReakitButton(props);

  const className = utils.useClassName({
    style: styles.Button,
    styleProps: props,
    prevClassName: boxProps.className
  });

  return { ...boxProps, ...buttonProps, className };
}

export function Button(props: ButtonProps) {
  const { use, children, ...restProps } = props;
  const buttonProps = useButtonProps(restProps);
  return (
    <utils.Element component={ReakitButton} use={use} htmlProps={buttonProps}>
      {children}
    </utils.Element>
  );
}
