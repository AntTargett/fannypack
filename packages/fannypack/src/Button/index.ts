import * as buttonStyles from './styles';

export * from './Button';
export { buttonStyles };
