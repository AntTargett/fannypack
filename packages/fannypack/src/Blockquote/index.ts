import Blockquote from './Blockquote';
import * as styledBlockquote from './styled';

export default Blockquote;

export * from './Blockquote';
export { default as Blockquote } from './Blockquote';

export { styledBlockquote };
