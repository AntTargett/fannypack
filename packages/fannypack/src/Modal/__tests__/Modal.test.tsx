import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Button } from '../../Button';
import { Blockquote } from '../../Blockquote';
import { Modal } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    function Component() {
      const modal = Modal.useState({ unstable_hiddenId: 'test' });
      return (
        <div>
          <Modal.Disclosure {...modal}>Toggle</Modal.Disclosure>
          <Modal {...modal} aria-label="test">
            Hello world
          </Modal>
        </div>
      );
    }
    const { baseElement } = render(<Component />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render correctly with non-modal modal', () => {
    function Component() {
      const modal = Modal.useState({ unstable_hiddenId: 'test' });
      return (
        <div>
          <Modal.Disclosure {...modal}>Toggle</Modal.Disclosure>
          <Modal {...modal} modal={false} aria-label="test">
            Hello world
          </Modal>
        </div>
      );
    }
    const { baseElement } = render(<Component />);
    expect(baseElement).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('use', () => {
    it('should render correctly', () => {
      function Component() {
        const modal = Modal.useState({ unstable_hiddenId: 'test' });
        return (
          <div>
            <Modal.Disclosure use={Button} {...modal}>
              Toggle
            </Modal.Disclosure>
            <Modal use={Blockquote} {...modal} aria-label="test">
              Hello world
            </Modal>
          </div>
        );
      }
      const { baseElement } = render(<Component />);
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Modal props', () => {
      const { result } = renderHook(() => {
        const modal = Modal.useState({ unstable_hiddenId: 'test' });
        return Modal.useProps(modal);
      });
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      function Component() {
        const modal = Modal.useState({ unstable_hiddenId: 'test' });
        return (
          <Modal {...modal}>
            {modalProps => (
              <div {...modalProps.htmlProps} aria-label="test">
                Hello world
              </div>
            )}
          </Modal>
        );
      }
      const { baseElement } = render(<Component />);
      expect(baseElement).toMatchSnapshot();
    });
  });
});

describe('overrides', () => {
  it('Modal.base should render correctly', () => {
    function Component() {
      const modal = Modal.useState({ unstable_hiddenId: 'test' });
      return (
        <Modal {...modal} aria-label="test" overrides={{ Modal: { base: { backgroundColor: 'red' } } }}>
          Hello world
        </Modal>
      );
    }
    const { baseElement } = render(<Component />);
    expect(baseElement).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('Modal.base should render correctly', () => {
    function Component() {
      const modal = Modal.useState({ unstable_hiddenId: 'test' });
      return (
        <Modal {...modal} aria-label="test">
          Hello world
        </Modal>
      );
    }
    const { baseElement } = render(<Component />, {
      theme: { Modal: { base: { backgroundColor: 'red' } } }
    });
    expect(baseElement).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    function Component() {
      const modal = Modal.useState({ unstable_hiddenId: 'test' });
      return (
        <Modal {...modal} aria-label="test">
          Hello world
        </Modal>
      );
    }
    const { baseElement } = render(<Component />, {
      theme: { Modal: { defaultProps: { className: 'test' } } }
    });
    expect(baseElement).toMatchSnapshot();
  });
});
