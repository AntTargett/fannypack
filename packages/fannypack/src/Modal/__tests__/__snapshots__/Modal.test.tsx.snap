// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition hook should return with Modal props 1`] = `
Object {
  "aria-modal": true,
  "className": "css-s4daff css-1pwrqjj css-h173ik hidden",
  "data-dialog": true,
  "hidden": true,
  "id": "test",
  "onAnimationEnd": [Function],
  "onKeyDown": [Function],
  "onTransitionEnd": [Function],
  "ref": [Function],
  "role": "dialog",
  "style": Object {
    "display": "none",
  },
  "tabIndex": -1,
  "unstable_wrap": [Function],
}
`;

exports[`composition render props should render correctly 1`] = `
<body>
  <div>
    <div
      aria-label="test"
    >
      Hello world
    </div>
  </div>
</body>
`;

exports[`composition use should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-8.emotion-8.emotion-8 {
  display: none;
}

.emotion-9 {
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #212121;
  fill: #212121;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  -webkit-transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  border: 1px solid #b3b3b3;
}

.emotion-0[disabled],
.emotion-0[aria-disabled="true"] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.emotion-0:focus {
  outline: unset;
  z-index: 2;
  box-shadow: gray 0px 0px 0px 1px, #a6a6a6 0px 0px 0px 3px;
  border-color: transparent;
}

.emotion-0:hover {
  background-color: #efefef;
}

.emotion-0:hover:active {
  background-color: #e3e3e3;
}

.emotion-6 {
  border-left: 4px solid #ccc;
  padding: 1rem;
}

.emotion-6:not(:last-child) {
  margin-bottom: 1rem;
}

<body>
  <div>
    <div>
      <button
        aria-controls="test"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="emotion-0 emotion-1 emotion-2 emotion-3 emotion-1 emotion-2"
        tabindex="0"
        type="button"
      >
        Toggle
      </button>
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <blockquote
      aria-label="test"
      aria-modal="true"
      class="emotion-6 emotion-1 emotion-8 emotion-9 emotion-1 emotion-8 hidden"
      data-dialog="true"
      hidden=""
      id="test"
      role="dialog"
      style="display: none;"
      tabindex="-1"
    >
      Hello world
    </blockquote>
  </div>
</body>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2.emotion-2.emotion-2 {
  display: none;
}

.emotion-0 {
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

<body>
  <div />
  <div
    class="__reakit-portal"
  >
    <div
      aria-label="test"
      aria-modal="true"
      class="emotion-0 emotion-1 emotion-2 hidden test"
      data-dialog="true"
      hidden=""
      id="test"
      role="dialog"
      style="display: none;"
      tabindex="-1"
    >
      Hello world
    </div>
  </div>
</body>
`;

exports[`overrides Modal.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2.emotion-2.emotion-2 {
  display: none;
}

.emotion-0 {
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

.emotion-0 {
  background-color: red;
}

<body>
  <div />
  <div
    class="__reakit-portal"
  >
    <div
      aria-label="test"
      aria-modal="true"
      class="emotion-0 emotion-1 emotion-2 hidden"
      data-dialog="true"
      hidden=""
      id="test"
      role="dialog"
      style="display: none;"
      tabindex="-1"
    >
      Hello world
    </div>
  </div>
</body>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-5.emotion-5.emotion-5 {
  display: none;
}

.emotion-3 {
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

<body>
  <div>
    <div>
      <button
        aria-controls="test"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="emotion-0 emotion-1 emotion-2"
        tabindex="0"
        type="button"
      >
        Toggle
      </button>
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      aria-label="test"
      aria-modal="true"
      class="emotion-3 emotion-1 emotion-5 hidden"
      data-dialog="true"
      hidden=""
      id="test"
      role="dialog"
      style="display: none;"
      tabindex="-1"
    >
      Hello world
    </div>
  </div>
</body>
`;

exports[`props should render correctly with non-modal modal 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-5.emotion-5.emotion-5 {
  display: none;
}

.emotion-3 {
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

<body>
  <div>
    <div>
      <button
        aria-controls="test"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="emotion-0 emotion-1 emotion-2"
        tabindex="0"
        type="button"
      >
        Toggle
      </button>
      <div
        aria-label="test"
        aria-modal="false"
        class="emotion-3 emotion-1 emotion-5 hidden"
        data-dialog="true"
        hidden=""
        id="test"
        role="dialog"
        style="display: none;"
        tabindex="-1"
      >
        Hello world
      </div>
    </div>
  </div>
</body>
`;

exports[`theming Modal.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2.emotion-2.emotion-2 {
  display: none;
}

.emotion-0 {
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

.emotion-0 {
  background-color: red;
}

<body>
  <div />
  <div
    class="__reakit-portal"
  >
    <div
      aria-label="test"
      aria-modal="true"
      class="emotion-0 emotion-1 emotion-2 hidden"
      data-dialog="true"
      hidden=""
      id="test"
      role="dialog"
      style="display: none;"
      tabindex="-1"
    >
      Hello world
    </div>
  </div>
</body>
`;
